src/java/org/jivesoftware/openfire/archive/ArchiveSearcher.java

```diff
@@ -351,6 +351,11 @@ private Collection<Conversation> databaseSearch(ArchiveSearch search) {
             // rows we need returned. The syntax is LIMIT [rows] OFFSET [offset]
             else if (DbConnectionManager.getDatabaseType() == DbConnectionManager.DatabaseType.postgresql) {
                 query.append(" LIMIT ").append(numResults).append(" OFFSET ").append(startIndex);
+            }
+			// SQL Server optimization: use the OFFSET command to tell the database how many
+            // rows we need returned. The syntax is OFFSET [offset] FETCH NEXT [rows] ROWS ONLY
+			else if (DbConnectionManager.getDatabaseType() == DbConnectionManager.DatabaseType.sqlserver) {
+                query.append(" OFFSET ").append(startIndex).append(" ROWS FETCH NEXT ").append(numResults).append(" ROWS ONLY ");
```

Comment:
The syntax doesn't match the comment.

If the code is correct, would line splitting make it more readable?

query
.append(" first "). append (variable)
.appear( " second ".append(variable).append("two")

(Forgive formatting, I'm on mobile which _insists_ I want English rather than Java)

Reply:
doesnt match? why?
startindex = offset and
numResults = count of rows..

so ... OFFSET startIndex ROWS FETCH NEXT numResults ROWS ONLY ... should be correct